chroma-agent/chroma_agent/device_plugins/linux_components/zfs.py

```diff
@@ -17,6 +20,237 @@
 from iml_common.lib.exception_sandbox import exceptionSandBox
 from iml_common.lib import util
 
+STORE_FILENAME = '/tmp/store.json'
+
+filter_empty = partial(filter, None)
+strip_lines = partial(map, lambda x: x.strip())
+
+
+def write_to_store(key, value, filename=STORE_FILENAME):
+    data = {}
+
+    try:
+        with open(filename, 'r') as f:
+            data = json.loads(f.read())
+    except IOError:
+        pass
+
+    daemon_log.info('write_to_store(): writing zfs data to %s. key: %s' % (filename, key))
+    # preserve other keys, only overwrite the key specified
+    data[key] = value
+    with open(filename, 'w') as f:
+        f.write(json.dumps(data))
+
+
+def read_from_store(key, filename=STORE_FILENAME):
```

Comment:
Seems like this could be more general and just return the entire file.

Then you can read the key off of it in a separate call.

This could replace the same call above in `write_to_store`.